File: frontend/Tekus.Web/src/app/core/paged-result.ts
export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface PagedQuery {
  pageNumber: number;
  pageSize: number;
  search?: string;
  sortBy?: string;
  sortDescending?: boolean;
}

File: backend/src/Tekus.Application/Common/Models/PagedResult.cs
namespace Tekus.Application.Common.Models
{
    public record PagedResult<T>(List<T> Items, int TotalCount, int PageNumber, int PageSize)
    {
        public int TotalPages => PageSize <= 0 ? 0 : (int)Math.Ceiling(TotalCount / (double)PageSize);
    }
}
